profile/endpoints/endpoint-domain.go

```diff
@@ -88,72 +88,65 @@ func (ep *EndpointDomain) String() string {
 	return ep.renderPPP(ep.OriginalValue)
 }
 
-func isValidEndpointDomain(value string) bool {
-	// Check for root domain
-	if value == "." {
-		return true
-	}
-
-	// Check for a "contains" value.
-	if containsRegex.MatchString(value) {
-		return true
-	}
-
-	// Remove special leading characters.
-	value = strings.TrimLeft(value, ".*")
-
-	// Check if value is now a valid domain.
-	return netutils.IsValidFqdn(value)
-}
-
 func parseTypeDomain(fields []string) (Endpoint, error) {
 	domain := fields[1]
-
 	ep := &EndpointDomain{
 		OriginalValue: domain,
 	}
 
-	// fix domain ending
+	// Fix domain ending.
 	switch domain[len(domain)-1] {
-	case '.':
-	case '*':
+	case '.', '*':
 	default:
 		domain += "."
 	}
 
-	// Check if domain is valid.
-	if !isValidEndpointDomain(domain) {
-		return nil, nil
-	}
-
-	// fix domain case
+	// Fix domain case.
 	domain = strings.ToLower(domain)
+	needValidFQDN := true
 
 	switch {
 	case strings.HasPrefix(domain, "*") && strings.HasSuffix(domain, "*"):
 		ep.MatchType = domainMatchTypeContains
-		ep.Domain = strings.Trim(domain, "*")
-		return ep.parsePPP(ep, fields)
+		ep.Domain = strings.TrimPrefix(domain, "*")
+		ep.Domain = strings.TrimSuffix(ep.Domain, "*")
+		needValidFQDN = false
 
 	case strings.HasSuffix(domain, "*"):
 		ep.MatchType = domainMatchTypePrefix
-		ep.Domain = strings.Trim(domain, "*")
-		return ep.parsePPP(ep, fields)
+		ep.Domain = strings.TrimSuffix(domain, "*")
+		needValidFQDN = false
+
+		// Prefix matching cannot be combined with zone matching
+		if strings.HasPrefix(ep.Domain, ".") {
+			return nil, nil
+		}
 
 	case strings.HasPrefix(domain, "*"):
 		ep.MatchType = domainMatchTypeSuffix
-		ep.Domain = strings.Trim(domain, "*")
-		return ep.parsePPP(ep, fields)
+		ep.Domain = strings.TrimPrefix(domain, "*")
+		needValidFQDN = false
 
 	case strings.HasPrefix(domain, "."):
 		ep.MatchType = domainMatchTypeZone
-		ep.Domain = strings.TrimLeft(domain, ".")
+		ep.Domain = strings.TrimPrefix(domain, ".")
 		ep.DomainZone = "." + ep.Domain
-		return ep.parsePPP(ep, fields)
 
 	default:
 		ep.MatchType = domainMatchTypeExact
 		ep.Domain = domain
-		return ep.parsePPP(ep, fields)
 	}
+
+	// Validate domain "content".
+	switch {
+	case needValidFQDN && !netutils.IsValidFqdn(ep.Domain):
+		return nil, nil
+	case !needValidFQDN && !allowedDomainChars.MatchString(ep.Domain):
+		return nil, nil
+	case strings.Contains(ep.Domain, ".."):
+		// The above regex does not catch double dots.
+		return nil, nil
+	}
```

Comment:
I'd recomment to not use switches for complex boolean cases because it's really hard to reason about them and using nested if-else blocks is way more easy to read.

IMHO using naked switches is awesome for single-conditions but as soon as one starts combining conditions with `&&` or `||` reading and understanding a switch can be hard. 

```suggestion
	if needValidFQDN {
		if !netutils.IsValidFqdn(ep.Domain) {
			return nil, nil
		}
	} else {
		if !allowedDomainChars.MatchString(ep.Domain) {
			return nil, nil
		}
	}
	
	if strings.Contains(ep.Domain, "..") {
		return nil, nil
	}
```